### PR TITLE
Bugfix for cart-tab-sync in multistore environment

### PR DESCRIPTION
**Pull request by cewald**
### Related Issues

Related to #3838 & #4083

This is a copy of #5711 rebased on `hotfix/v1.12.3` branch.
 
### Short Description of the PR

The multi-tab cart-sync feature wouldn't work in a multistore-environment with disabled `storeViews.commonCache` feature as the keys in local-storage are prefixed with the store-code like `de-shop/cart/current-cart`.

### Pull Request Checklist

- [x] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [x] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

**Review by Fifciu — CHANGES_REQUESTED** [review]
Thanks for rapid reaction, I see you also refacored condition a bit which is great. Now it is much more readable. However, I have to ask you to move `checkMultistoreKey` function above `getItemsFromStorage` to respect our linter.

**Comment by cewald** [comment]
Sorry, my fault – it's done.

File: core/modules/cart/helpers/syncCartWhenLocalStorageChange.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+import { currentStoreView } from '@vue-storefront/core/lib/multistore'
+
+function checkMultistoreKey (key: string, path: string): boolean {
+  const { multistore, commonCache } = storeViews
+  if (!multistore || (multistore && commonCache)) return key === path
+  return key === `${currentStoreView().storeCode}-${path}`
+}
 
 function getItemsFromStorage ({ key }) {
   const value = JSON.parse(localStorage[key])
```
Comment (cewald) [inline core/modules/cart/helpers/syncCartWhenLocalStorageChange.ts:12]: We still need to fetch the correct key in `JSON.parse(localStorage[key])` when multistore is enabled.

Comment (cewald) [reply core/modules/cart/helpers/syncCartWhenLocalStorageChange.ts]: I've moved it into the if-else clause.

**Comment by Fifciu** [comment]
Please resolve conflict with the changelog and we could merge it

**Review by Fifciu — APPROVED** [review]

**Comment by Fifciu** [comment]
Great job, thanks!

**Pull request merged**